python/ray/dataframe/dataframe.py

```diff
@@ -2328,50 +2329,53 @@ def iterrows(self):
         """Iterate over DataFrame rows as (index, Series) pairs.
 
         Note:
-            Generators can't be pickeled so from the remote function
+            Generators can't be pickled so from the remote function
             we expand the generator into a list before getting it.
             This is not that ideal.
 
         Returns:
             A generator that iterates over the rows of the frame.
         """
-        def update_iterrow(series, i):
-            """Helper function to correct the columns + name of the Series."""
-            series.index = self.columns
-            series.name = list(self.index)[i]
-            return series
+        index_iter = iter([self._row_metadata.partition_series(i).index
```

Comment:
Prefer `index_iter = (self._row_metadata.partition_series(i).index for i in range(len(self._row_partitions)))`